From 6b9170887e1b912b657dab4597f8b44ae4dbdf50 Mon Sep 17 00:00:00 2001
From: Stefan Wahren <stefan.wahren@i2se.com>
Date: Wed, 16 Nov 2016 17:52:06 +0000
Subject: [PATCH] ARM: bcm2835: Fix names for the Raspberry Pi GPIO lines

There are some differences between the schematics and the official firmware
DTS [1]. So based on these additional information the following has been
changed:

* use consistent "CAM_GPIO1" for camera LED
* use consistent "CAM_GPIO0" for camera shutdown
* add "USB_LIMIT" for USB current limit (0=600mA, 1=1200mA)

[1] - https://github.com/raspberrypi/firmware/blob/master/extra/dt-blob.dts

Signed-off-by: Stefan Wahren <stefan.wahren@i2se.com>
Signed-off-by: Eric Anholt <eric@anholt.net>
---
 arch/arm/boot/dts/bcm2835-rpi-a-plus.dts | 4 ++--
 arch/arm/boot/dts/bcm2835-rpi-a.dts      | 4 ++--
 arch/arm/boot/dts/bcm2835-rpi-b-plus.dts | 2 +-
 arch/arm/boot/dts/bcm2835-rpi-b.dts      | 4 ++--
 4 files changed, 7 insertions(+), 7 deletions(-)

--- a/arch/arm/boot/dts/bcm2835-rpi-a-plus.dts
+++ b/arch/arm/boot/dts/bcm2835-rpi-a-plus.dts
@@ -63,13 +63,13 @@
 			  "SCL0",
 			  "NC", /* GPIO30 */
 			  "NC", /* GPIO31 */
-			  "NC", /* GPIO32 */
+			  "CAM_GPIO1", /* GPIO32 */
 			  "NC", /* GPIO33 */
 			  "NC", /* GPIO34 */
 			  "PWR_LOW_N", /* GPIO35 */
 			  "NC", /* GPIO36 */
 			  "NC", /* GPIO37 */
-			  "NC", /* GPIO38 */
+			  "USB_LIMIT", /* GPIO38 */
 			  "NC", /* GPIO39 */
 			  "PWM0_OUT", /* GPIO40 */
 			  "CAM_GPIO0", /* GPIO41 */
--- a/arch/arm/boot/dts/bcm2835-rpi-a.dts
+++ b/arch/arm/boot/dts/bcm2835-rpi-a.dts
@@ -29,7 +29,7 @@
 			  "SDA1",
 			  "SCL1",
 			  "GPIO_GCLK",
-			  "CAM_CLK",
+			  "CAM_GPIO1",
 			  "LAN_RUN",
 			  "SPI_CE1_N",
 			  "SPI_CE0_N",
@@ -52,7 +52,7 @@
 			  "GPIO24",
 			  "GPIO25",
 			  "NC", /* GPIO26 */
-			  "CAM_GPIO",
+			  "CAM_GPIO0",
 			  /* Binary number representing build/revision */
 			  "CONFIG0",
 			  "CONFIG1",
--- a/arch/arm/boot/dts/bcm2835-rpi-b-plus.dts
+++ b/arch/arm/boot/dts/bcm2835-rpi-b-plus.dts
@@ -71,7 +71,7 @@
 			  "PWR_LOW_N", /* GPIO35 */
 			  "NC", /* GPIO36 */
 			  "NC", /* GPIO37 */
-			  "NC", /* GPIO38 */
+			  "USB_LIMIT", /* GPIO38 */
 			  "NC", /* GPIO39 */
 			  "PWM0_OUT", /* GPIO40 */
 			  "CAM_GPIO0", /* GPIO41 */
--- a/arch/arm/boot/dts/bcm2835-rpi-b.dts
+++ b/arch/arm/boot/dts/bcm2835-rpi-b.dts
@@ -30,7 +30,7 @@
 			  "SDA1",
 			  "SCL1",
 			  "GPIO_GCLK",
-			  "CAM_CLK",
+			  "CAM_GPIO1",
 			  "LAN_RUN",
 			  "SPI_CE1_N",
 			  "SPI_CE0_N",
@@ -53,7 +53,7 @@
 			  "GPIO24",
 			  "GPIO25",
 			  "NC", /* GPIO26 */
-			  "CAM_GPIO",
+			  "CAM_GPIO0",
 			  /* Binary number representing build/revision */
 			  "CONFIG0",
 			  "CONFIG1",
